rust/src/measurements/laplace_threshold/mod.rs

```diff
@@ -57,39 +48,47 @@ pub fn make_base_laplace_threshold<TK, TV>(
 >
 where
     TK: Hashable,
-    TV: Float + SampleDiscreteLaplaceZ2k,
-    i32: ExactIntCast<TV::Bits>,
+    TV: Float + RoundCast<FBig>,
+    u32: ExactIntCast<TV::Bits>,
+    FBig: TryFrom<TV, Error = ConversionError>,
     (MapDomain<AtomDomain<TK>, AtomDomain<TV>>, L1Distance<TV>): MetricSpace,
 {
     if input_domain.value_domain.nullable() {
-        return fallible!(FailedFunction, "values must be non-null");
+        return fallible!(MakeMeasurement, "values must be non-null");
     }
 
     if threshold < TV::zero() {
-        return fallible!(FailedFunction, "threshold must be non-negative");
+        return fallible!(MakeMeasurement, "threshold must be non-negative");
     }
 
     if scale < TV::zero() {
-        return fallible!(FailedFunction, "scale must be non-negative");
+        return fallible!(MakeMeasurement, "scale must be non-negative");
     }
 
     let _2 = TV::exact_int_cast(2)?;
-    let (k, relaxation) = get_discretization_consts(k)?;
 
-    // actually reject noisy values below threshold + relaxation, to account for discretization
-    let true_threshold = threshold.inf_add(&relaxation)?;
+    let f_scale = FBig::try_from(scale)?;
+    let f_threshold = FBig::try_from(threshold)?;
+
     Measurement::new(
         input_domain,
         Function::new_fallible(move |data: &HashMap<TK, TV>| {
             data.clone()
                 .into_iter()
                 // noise output count
-                .map(|(key, v)| TV::sample_discrete_laplace_Z2k(v, scale, k).map(|v| (key, v)))
-                // only keep keys with values gte threshold
-                .filter(|res| {
-                    res.as_ref()
-                        .map(|(_k, v)| v >= &true_threshold)
-                        .unwrap_or(true)
+                .map(|(key, value)| {
+                    // value should be non-null, but if it is, attempt to degrade gracefully
+                    let f_value = FBig::try_from(value).unwrap_or(FBig::ZERO);
```

Comment:
I made another slight adjustment to make this even more graceful- if the count is NaN it will be filtered. 

In general, we make the assumption that data is in the input domain. If data is not, then if we can keep the adversary from finding out without significantly complicating the code, it's best to do so. Raising an error could constitute a much more severe privacy violation. If the analyst's threat model allows it, there is a separate utility to check that data is in the input domain that that they can make use of.